File: whistleblower/workflows-java/src/test/java/net/corda/examples/whistleblower/NodeDriver.java
package net.corda.examples.whistleblower;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableSet;
import net.corda.core.identity.CordaX500Name;
import net.corda.testing.driver.DriverParameters;
import net.corda.testing.driver.NodeParameters;
import net.corda.testing.node.User;

import java.util.List;
import java.util.concurrent.ExecutionException;

import net.corda.testing.driver.Driver;

/**
 * Allows you to run your nodes through an IDE (as opposed to using deployNodes). Do not use in a production
 * environment.
 */
public class NodeDriver {
    public static void main(String[] args) {
        List<User> rpcUsers = ImmutableList.of(new User("user1", "test", ImmutableSet.of("ALL")));

        Driver.driver(new DriverParameters().withWaitForAllNodesToFinish(true).withStartNodesInProcess(true), driverDSL -> {

            try {
                driverDSL.startNode(new NodeParameters().withProvidedName(new CordaX500Name("PartyA", "London", "GB")).withRpcUsers(rpcUsers)).get();
                driverDSL.startNode(new NodeParameters().withProvidedName(new CordaX500Name("PartyB", "New York", "US")).withRpcUsers(rpcUsers)).get();
            } catch (InterruptedException | ExecutionException e) {
                e.printStackTrace();
            }
            return null;
        });
    }

}

File: whistleblower/src/test/kotlin/com/whistleblower/NodeDriver.kt
package com.whistleblower

import net.corda.core.identity.CordaX500Name
import net.corda.core.utilities.getOrThrow
import net.corda.testing.driver.DriverParameters
import net.corda.testing.driver.driver
import net.corda.testing.node.User

/**
 * This file is exclusively for being able to run your nodes through an IDE (as opposed to using deployNodes)
 * Do not use in a production environment.
 */
fun main(args: Array<String>) {
    // No permissions required as we are not invoking flows.
    val user = User("user1", "test", permissions = setOf("ALL"))
    driver(DriverParameters(isDebug = true, startNodesInProcess = true, waitForAllNodesToFinish = true)) {
        val (nodeA, nodeB, nodeC, nodeD) = listOf(
                startNode(providedName = CordaX500Name("BraveEmployee", "Nairobi", "KE"), rpcUsers = listOf(user)).getOrThrow(),
                startNode(providedName = CordaX500Name("TradeBody", "Kisumu", "KE"), rpcUsers = listOf(user)).getOrThrow(),
                startNode(providedName = CordaX500Name("GovAgency", "Mombasa", "KE"), rpcUsers = listOf(user)).getOrThrow(),
                startNode(providedName = CordaX500Name("BadCompany", "Eldoret", "KE"), rpcUsers = listOf(user)).getOrThrow())

        startWebserver(nodeA)
        startWebserver(nodeB)
        startWebserver(nodeC)
        startWebserver(nodeD)
    }
}
